### PR TITLE
[feat] Handle trigger of revision mode

### PR DESCRIPTION
**Pull request by joassouza**
This PR implements the revision mode logic. This is how the things are wired...
## Entering on revision mode
1. User clicks on revision mode button (WEB)
 1.1 Web changes the attribute :revision_mode on the document model
 1.2 Web changes "revisionMode" param to `true` (it has no effect yet, need to reload)
 1.3 web sends a message to Editor API to prepare to revision mode
2. Editor get the message then clears the authorship attribute (This attribute is used to show the texts edited)
  2.1. Editor checks from time to time if the authorship attribute is cleared 
3. As soon we clean all author attribute on script, editor sends a message informing is ready to enter on revision mode
4. Web listen to this message and reload the editor so the font color takes effect

#### Most of the things are done on the Web side
1. User clicks on revision mode button (WEB)
2. We change the param "revisionMode" to false
3. Reload the editor
4. The hook handles the font color as the black (default one). All the magic is done on the hook `aceSetAuthorStyle` 
### This PR
- [x] Trigger RM
- [x] Handle globally the revision mode

### Future Work
- [ ] Show warning 
- [ ] Optimize performance when clear authorship

### Observations
1. We removed the diff generation API call, due to serious security bugs
2. This is the WIP web PR https://github.com/storytouch/web/pull/1048/files

https://trello.com/c/qZQKuNuv/2199-revision-mode-10

### CODE REVIEW

File: static/js/api.js [hunk 1]
```diff
@@ -96,6 +90,14 @@ epCompareApi.prototype.triggerDataChanged = function(diffData) {
   this._triggerEvent(message);
 }
 
+epCompareApi.prototype.triggerEditorIsReadyToEnterOnRevisionMode = function(diffData) {
```
Comment (ingoncalves) [inline static/js/api.js:93]: `diffData` não é usado aqui. Seria melhor remover da assinatura?

File: static/js/revisionMode.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    }
+  }
+}
+
+revisionMode.prototype._getAuthorsClasses = function() {
+  var authors = clientVars.collab_client_vars.historicalAuthorData
+  return _.map(_.keys(authors), this._getAuthorClasses, '').join(',');
+}
+
+// that's what Etherpad does to create author classes
```
Comment (ingoncalves) [inline static/js/revisionMode.js:52]: Seria legal colocar aqui nos comentários um exemplo de `entrada -> saída` dessa função. Esses `replaces` assustam um pouco hahaha.

File: static/tests/frontend/specs/calculateHunks.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-describe('ep_compare - create hunks information', function() {
+describe.skip('ep_compare - create hunks information', function() {
```
Comment (ingoncalves) [inline static/tests/frontend/specs/calculateHunks.js:1]: Estes `.skip` são propositais ou esqueceu de remover?

Comment (joassouza) [reply static/tests/frontend/specs/calculateHunks.js:1]: intencional...O código continua aí mas a gente não vai mais usar desse jeito. Muita coisa vai mudar

File: static/js/revisionMode.js (hunk 2 above)
Comment (joassouza) [reply static/js/revisionMode.js:52]: vdd!

File: static/js/api.js (hunk 1 above)
Comment (joassouza) [reply static/js/api.js:93]: copy/paste! 🤦

**Pull request merged**